src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -181,68 +170,85 @@ bool ThetaStarPathPlanner::updateVertex(CellCoordinate pCurr, CellCoordinate pNe
 }
 
 // top level function
-std::optional<std::vector<Point>> ThetaStarPathPlanner::findPath(const Point &start,
-                                                                 const Point &destination)
+std::variant<std::monostate, Curve, std::vector<Point>> ThetaStarPathPlanner::findPath(
+    const Point &start, const Point &dest, const Field &field,
+    const std::vector<Obstacle> &obstacles)
 {
-    CellCoordinate src, dest;
+    obstacles_ = obstacles;
+    std::variant<std::monostate, Curve, std::vector<Point>> retval;
+    CellCoordinate src_coord, dest_coord;
+
```

Comment:
In general, member variables should be set in the constructor initialization list when possible. I think most of our members here can be.

https://stackoverflow.com/questions/926752/why-should-i-prefer-to-use-member-initialization-list (and also just to be consistent with the rest of the codebase).

Reply:
#814

Reply:
Gonna assume by that you mean "out of scope". Agreed!